packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -719,10 +720,27 @@ export const RemoteServerFactory = {
 
         app.use(
             slasPrivateProxyPath,
+            (req, res, next) => {
+                // Check if the request should be blocked before it reaches the proxy
+                // We run this outside of the proxy middleware because modifying the response
+                // to send a 403 in the proxy causes issues with the response interceptor.
+                if (
+                    !req.path?.match(options.slasApiPath) ||
+                    req.path?.match(/\/oauth2\/trusted-system/)
+                ) {
+                    const message = `Request to ${req.path} is not allowed through the SLAS Private Client Proxy`
+                    logger.error(message)
+                    return res.status(403).json({
+                        message: message
+                    })
+                }
+                next()
+            },
             createProxyMiddleware({
                 target: options.slasTarget,
                 changeOrigin: true,
                 pathRewrite: {[slasPrivateProxyPath]: ''},
+                selfHandleResponse: true,
```

Comment:
This needs to be true in order to use the response interceptor, as documented by http-proxy-middleware docs.